src/conf_mode/containers.py

```diff
@@ -0,0 +1,218 @@
+            if 'image' not in container_config:
+                raise ConfigError(f'Image for container "{cont}" is required!')
+
+            # If 'allow-host-networks' or 'network' not set.
+            if 'allow-host-networks' not in container_config and 'network' not in container_config:
+                raise ConfigError(f'"Network" or "allow-host-networks" for container "{cont}" is required!')
+
+            # If set both parameters for networks (host and user-defined). We require only one.
+            if 'allow-host-networks' in container_config and 'network' in container_config:
+                raise ConfigError(f'"allow-host-networks" and "network" for "{cont}" cannot be both configured at the same time!')
+
+    # Add new network
+    if 'network' in container:
+        for net in container['network']:
+            # If ipv4-prefix not defined for user-defined network
+            if 'ipv4-prefix' not in container['network'][net]:
+                raise ConfigError(f'IPv4 prefix for network "{net}" is required!')
+
+    return None
+
+def generate(container):
+    # bail out early - looks like removal from running config
+    if not container:
+        return None
+
+    render(config_containers_registry, 'containers/registry.tpml', container)
+
+    return None
+
+def apply(container):
+    # Delete old networks if needed
+    if 'net_remove' in container:
+        for net in container['net_remove']:
+            call(f'podman network rm {net} > /dev/null 2>&1')
+
+    # Delete old containers if needed. We can't delete running container
+    if 'container_remove' in container:
+        for name in container['container_remove']:
+            call(f'podman stop {name} > /dev/null 2>&1')
+            call(f'podman rm {name} > /dev/null 2>&1')
+            print(f'Container {name} deleted')
+
+    # Add network
+    if 'network' in container:
+        for net in container['network']:
+            # Check if the network has already been created
+            if ctnr_network_exists(net) is False:
+                prefix = container['network'][net]['ipv4-prefix']
+                if container['network'][net]['ipv4-prefix']:
+                    # Create user-defined network
+                    try:
+                        call(f'podman network create {net} --subnet={prefix} > /dev/null 2>&1')
+                    except:
+                        print(f'Can\'t add network {net}')
+
+    # Add container
+    if 'name' in container:
+        for name in container['name']:
+            # Check if the container has already been created
+            #if len(cmd(f'podman ps -a --filter "name=^{name}$" -q')) == 0:
+            if container_exists(name) is False:
+                image = container['name'][name]['image']
+
+                if 'allow-host-networks' in container['name'][name]:
+                    try:
+                        call(f'podman run -dit --name {name} --net host {image} > /dev/null 2>&1')
```

Comment:
plese use cmd() instead of call(), as cmd will autofail on error. Then you can also get rid of the try/except.

Reply:
> plese use cmd() instead of call(), as cmd will autofail on error. Then you can also get rid of the try/except.

Done. Replaced all "call => cmd"